docs/PARALLEL_CHECKSUMS.md

```diff
@@ -0,0 +1,32 @@
+# PARALLEL CHECKSUMS
+
+## sha2-256-parallel-8
+
+### 0xb518
+
+This variant of sha2-256 is designed to enable very large files
+to be uploaded and hashed in parallel using fixed size blocks
+(8MiB to start with), with the final result being a "top_hash"
+that is an order-independent hash of the individual hashes.
+
+The algorithm has an upper limit of 10,000 hashes;
+if the file is larger than 80,000 MiB,\
+it will double the block size until the number of blocks
+is under that limit.
+
+* If the file is less than 8 MiB,
+  the result will be identical to "serial" sha2-256.
+
+* If the file is exactly 8 MiB
+  the result will be the hash of the file hashed with a zero-size block.
```

Comment:
@drernie: I don't think I understand "hashed with a zero-size block". It's good to call out this corner case, but I'd describe it as a hash of one individual hash, or something along those words. I.e., the multi-hash algorithm is used for files ≥8MB, even though >8MB would have been more logical.